File: cases/taint/go/same-object-field-positive/same_object_field_positive.go
package dataflowbench

type Holder struct {
	Tainted int
	Clean   int
}

func dfb_source() int { // DFB-SOURCE: same-object-field-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: same-object-field-sink

func run() {
	holder := Holder{}
	holder.Tainted = dfb_source() // DFB-WITNESS: same-object-field-store
	holder.Clean = 0
	dfb_sink(holder.Tainted)
}

File: cases/taint/csharp/same-object-field-positive/SameObjectFieldPositive.cs
namespace DataFlowBench;

static class SameObjectFieldPositive
{
    sealed class Holder
    {
        public int Tainted;
        public int Clean;
    }

    static int dfb_source() // DFB-SOURCE: same-object-field-input
    {
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: same-object-field-sink

    static void Run()
    {
        Holder holder = new Holder();
        holder.Tainted = dfb_source(); // DFB-WITNESS: same-object-field-store
        holder.Clean = 0;
        dfb_sink(holder.Tainted);
    }
}
